File: Nice_Pairs_.py
for i in range(int(input())):
    n = int(input())
    s = input()
    count = 0
    for j in range(n - 1):
        for k in range(j + 1, n):
            if abs(int(s[j]) - int(s[k])) == k - j:
                count += 1
    print(count)

File: Nice_Pairs_.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    cin >> t;
    while(t--)
    {
        int n;
        cin >> n;
        string s;
        cin >> s;
        int count = 0;
        for(int i = 0; i < n - 1; i++)
        {
            for(int j = i + 1; j < n; j++)
            {
                if(abs(int(s[j]) - int(s[i])) == j - i)
                {
                    count += 1;
                }
            }
        }
        cout << count << endl;
    }
    return 0;
}